--- yy条件素数筛.wasm.wat	2026-09-20 03:49:46
+++ yy计数邻接原型.wat	2026-09-20 03:56:04
@@ -107444,8 +107444,7 @@
 local.get $i2
 local.get $i1
 i64.gt_s
-local.set $p4
-local.get $p4
+
 if (result (ref null eq))
 local.get $i3 call $box
 else
@@ -107460,8 +107459,7 @@
 local.get $i2
 local.tee $j i64.const 4294967295 i64.gt_u if unreachable end local.get $j i32.wrap_i64
 array.get_u $octs i64.extend_i32_u
-local.set $i6
-local.get $i6 i32.wrap_i64
+ i32.wrap_i64
 i32.const 0 i32.ne i64.extend_i32_u
 local.get $i3
 i64.add local.set $i8
